src/tangd.c

```diff
@@ -273,8 +301,8 @@ main(int argc, char *argv[])
     jwkdir = argv[optind++];
 
     if (listen == 0) { /* process one-shot query from stdin */
-	return process_request(jwkdir, STDIN_FILENO);
+        return process_request(jwkdir, STDIN_FILENO, endpoint);
```

Comment:
Instead of passing `endpoint` here and  later on in `run_service()`, it would probably be better to adjust `s_dispatch` for the endpoint,  when required.

Reply:
Not sure what you mean with this. `run_service()` calls `spawn_process()` and `spawn_process()` calls `process_request()`. This last function adapts the s_dispatch with the endpoint, so it needs user provided endpoint to adjust it